vector/src/test/java/im/vector/app/features/location/LocationDataTest.kt

```diff
@@ -62,14 +64,14 @@ class LocationDataTest {
     }
 
     @Test
-    fun genericLocationTest() {
+    fun selfLocationTest() {
         val contentWithNullAsset = MessageLocationContent(body = "", geoUri = "", locationAsset = null)
-        contentWithNullAsset.isGenericLocation() shouldBeEqualTo(false)
+        contentWithNullAsset.isSelfLocation().shouldBeTrue()
 
         val contentWithNullAssetType = MessageLocationContent(body = "", geoUri = "", locationAsset = LocationAsset(type = null))
-        contentWithNullAssetType.isGenericLocation() shouldBeEqualTo(true)
+        contentWithNullAssetType.isSelfLocation().shouldBeTrue()
```

Comment:
Weird, there is a logic change here. But the newer version seems fine. So the previous test was wrong?

Reply:
> If m.asset is missing from the location's content the client should render it as m.self as that will be the most common use case. Otherwise, if it's not missing but the type is invalid or unknown the client should attempt to render it as a generic location.

Yes, MSC confused me. But last one seems ok. I will also check with other clients.

Reply:
OK, so the correct test should be `contentWithNullAssetType.isSelfLocation().shouldBeFalse()`, in this case no?

Reply:
We just discussed internally and the missing `type` is `m.asset`.